src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey <T>(item: T[], key: keyof T): GroupsMap<T> {
+  const result: GroupsMap<T> = {};
```

Comment:
It is better to avoid naming variables `result`. Try to specify what exactly the result is